src/rebar_compiler_dag.erl

```diff
@@ -224,6 +224,7 @@ compile_order(G, AppDefs) ->
     AppPaths = prepare_app_paths(AppDefs),
     compile_order(Edges, AppPaths, #{}).
 
+-dialyzer({nowarn_function, maybe_store/5}).
```

Comment:
what were the issues with these?

Reply:
Matching with an opaque structure. I looked at OTP and there's no way to get those elements from the record, in any case.

Reply:
ah yeah, the DAG exploding code breaks promises. I'd have thought maybe_store/5 itself would be okay though.

Reply:
> I'd have thought maybe_store/5 itself would be okay though.

It wasn't, because of the `G`, passed further to `store_dag`.

Reply:
Are these (`nowarn_...`) OK with you? Or should I change anything?

Reply:
They're fine yeah. Might want to trail them with a `% optimized digraph usage breaks opacity` comment to explain why.

Reply:
I'd chosen a different comment, but will override with yours.

Reply:
Better comment is in place.